### PR TITLE
fix up translation

### PR DESCRIPTION
**Pull request by jasonren0403**
fix up missing translation in web/security/secure_contexts

### CODE REVIEW

File: files/zh-cn/web/security/secure_contexts/index.md [hunk 1]
```diff
@@ -2,53 +2,55 @@
 title: Secure contexts
```
Comment (jasonren0403) [inline files/zh-cn/web/security/secure_contexts/index.md:2]:
```suggestion
title: 安全上下文
```

File: files/zh-cn/web/security/secure_contexts/index.md [hunk 2]
```diff
@@ -1,54 +1,56 @@
 ---
-title: Secure contexts
+title: 安全上下文
 slug: Web/Security/Secure_Contexts
 ---
-**安全上下文**是 `Window` 与 `Worker` 中的概念满足了最低标准的身份验证和机密性。许多 Web APIs 的访问仅能在安全上下文中. 安全上下文的主要目标是防止 [MITM attackers](https://zh.wikipedia.org/wiki/man-in-the-middle_attack) 强大的 APIs 被坏人利用。
+**安全上下文**是 `Window` 与 `Worker` 中满足了最低标准的身份验证和机密性的概念。许多 Web API 仅能在安全上下文中访问。安全上下文的主要目标是防止[中间人攻击者](https://zh.wikipedia.org/wiki/man-in-the-middle_attack)访问强大的接口，从而导致更加严重的破坏。
```
Comment (jasonren0403) [inline files/zh-cn/web/security/secure_contexts/index.md:5]:
```suggestion
**安全上下文**是 `Window` 与 `Worker` 中满足了最低标准的身份验证和机密性的概念。许多 Web API 仅能在安全上下文中访问。安全上下文的主要目标是防止[中间人攻击者](https://zh.wikipedia.org/wiki/中间人攻击)访问强大的接口，从而导致更加严重的破坏。
```

File: files/zh-cn/web/security/secure_contexts/index.md [hunk 3]
```diff
@@ -1,54 +1,56 @@
...
 
-For example, even for a document delivered over TLS within an {{HTMLElement("iframe")}}, its context is **not** considered secure if it has an ancestor that was not also delivered over TLS.
+例如，在 {{HTMLElement("iframe")}} 中的文档即使通过 TLS 进行传输，如果它有一个父节点没有通过 TLS 进行传输，其上下文也**不会**视为安全。
 
-However, it’s important to note that if a non-secure context causes a new window to be created (with or without specifying [noopener](/en-US/docs/Web/API/Window/open#noopener)), then the fact that the opener was insecure has no effect on whether the new window is considered secure. That’s because the determination of whether or not a particular document is in a secure context is based only on considering it within the top-level browsing context with which it is associated — and not whether a non-secure context happened to be used to create it.
+然而，如果不安全上下文导致新窗口的建立（无论有无指定 [noopener](/zh-CN/docs/Web/API/Window/open) 属性），那么引起开启行为发生的目标不安全的事实不会影响新窗口的安全性。这是因为，确定一个特定的文档是否在安全上下文中，只基于在与之相关的顶层浏览上下文中考虑它——而不是在非安全上下文中碰巧创建了它。
```
Comment (yin1999) [inline files/zh-cn/web/security/secure_contexts/index.md:21]: 这句话还有问题，请修改

File: files/zh-cn/web/security/secure_contexts/index.md [hunk 4]
```diff
@@ -1,54 +1,56 @@
...
+## 特性检测
+
+页面可以通过暴露在全局范围的 {{domxref("isSecureContext")}} 布尔属性值真假来判断它是否处于安全上下文中。
 
 ```js
 if (window.isSecureContext) {
-  // Page is a secure context so service workers are now available
-  navigator.serviceWorker.register("/offline-worker.js").then(function () {
-    ...
+  // 页面在安全上下文中，所以 service workers 也会可用
```
Comment (yin1999) [inline files/zh-cn/web/security/secure_contexts/index.md:38]:
```suggestion
  // 页面在安全上下文中，所以 service worker 可用
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/security/secure_contexts/index.md (hunk 3 above)
Comment (jasonren0403) [reply files/zh-cn/web/security/secure_contexts/index.md:21]:
```suggestion
然而，需要引起注意的是，如果某个不安全上下文导致建立了新窗口（无论是否指定 [noopener](/zh-CN/docs/Web/API/Window/open#noopener) 参数），那么引起开启行为发生的目标不安全的事实不会影响新窗口的安全性。这是因为，确定一个特定的文档是否在安全上下文中，仅仅考虑与之相关的顶层浏览上下文中，而不是是否碰巧使用了非安全的上下文来创建它。
```

**Review by yin1999 — APPROVED** [review]
Thanks @jasonren0403 👍

**Pull request merged**